includes/oauth/class-fivetran-connection.php

```diff
@@ -0,0 +1,237 @@
+<?php
+/**
+ * Connection to Google services.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack;
+
+use WP_Error;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Main class.
+ */
+class Fivetran_Connection {
+	/**
+	 * Constructor.
+	 *
+	 * @codeCoverageIgnore
+	 */
+	public function __construct() {
+		add_action( 'rest_api_init', [ $this, 'register_api_endpoints' ] );
+	}
+
+	/**
+	 * Register the endpoints.
+	 *
+	 * @codeCoverageIgnore
+	 */
+	public function register_api_endpoints() {
+		register_rest_route(
+			NEWSPACK_API_NAMESPACE,
+			'/oauth/fivetran',
+			[
+				'methods'             => \WP_REST_Server::CREATABLE,
+				'callback'            => [ $this, 'api_modify_connector' ],
+				'permission_callback' => [ $this, 'api_permissions_check' ],
+				'args'                => [
+					'connector_id' => [
+						'required'          => true,
+						'sanitize_callback' => 'sanitize_text_field',
+					],
+					'paused'       => [
+						'sanitize_callback' => 'rest_sanitize_boolean',
+					],
+				],
+			]
+		);
+		register_rest_route(
+			NEWSPACK_API_NAMESPACE,
+			'/oauth/fivetran',
+			[
+				'methods'             => \WP_REST_Server::READABLE,
+				'callback'            => [ $this, 'api_get_fivetran_connection_status' ],
+				'permission_callback' => [ $this, 'api_permissions_check' ],
+			]
+		);
+		register_rest_route(
+			NEWSPACK_API_NAMESPACE,
+			'/oauth/fivetran/(?P<provider>[\a-z]+)',
+			[
+				'methods'             => \WP_REST_Server::EDITABLE,
+				'callback'            => [ $this, 'api_create_connection' ],
+				'permission_callback' => [ $this, 'api_permissions_check' ],
+				'args'                => [
+					'service' => [
+						'required'          => true,
+						'sanitize_callback' => 'sanitize_text_field',
+					],
+				],
+			]
+		);
+	}
+
+	/**
+	 * Get the Fivetran proxy URL.
+	 *
+	 * @param string $path Path to append to base URL.
+	 */
+	private static function get_proxy_url( $path = '' ) {
+		if ( ! self::is_fivetran_configured() ) {
+			return false;
+		}
+		return add_query_arg(
+			[
+				'wpcom_access_token' => urlencode( base64_encode( WPCOM_OAuth::get_access_token() ) ),
```

Comment:
Must check if value returned from `WPCOM_OAuth::get_access_token()` is not a `WP_Error`